src/explore-education-statistics-common/src/components/ContentSectionIndex.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+
+interface IndexResult {
+  element: Element;
+  location: string;
+}
+
+interface Props {
+  fromId: string;
+}
+
+interface State {
+  indexResults: IndexResult[];
+  indexComplete: boolean;
+}
+
+class ContentSectionIndex extends Component<Props> {
+  public state: State = {
+    indexResults: [],
+    indexComplete: false,
+  };
+
+  public componentDidMount() {
+    const { fromId } = this.props;
+    this.index(fromId);
```

Comment:
We need this in a `componentDidUpdate` as well, in case the `fromId` prop is changed by the parent. It should also reset the state of `indexComplete` and `indexResults` beforehand.